### PR TITLE
Major version changes - ios

### PR DESCRIPTION
**Pull request by jjliu15**
## Description

Upgrades iOS to 8.4.0. This only has changes to iOS - I will send out another PR to update the flutter code to be compatible with the new Android and iOS implementations.

- Renames `DFP` to `GAM`
- Removes `Unified` from `UnifiedNativeAd`
- Support updated delegates and full screen APIs
- Split unit tests into format specific files

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`).
This will ensure a smooth and quick review process. Updating the `pubspec.yaml` and changelogs is not required.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [x] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [ ] No, this is *not* a breaking change.

### CODE REVIEW

File: packages/google_mobile_ads/ios/Tests/FLTNativeAdTest.m [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+  GADAdLoader *mockLoader = OCMPartialMock([ad adLoader]);
+  OCMStub([mockNativeAd adLoader]).andReturn(mockLoader);
+  [mockNativeAd load];
+  
+  OCMVerify([mockLoader loadRequest:[OCMArg checkWithBlock:^BOOL(id obj) {
+                          GADRequest *requestArg = obj;
+                          return [requestArg.keywords isEqualToArray:@[ @"apple" ]];
+                        }]]);
+
+  // Check that nil is used instead of null when customOptions is Null
```
Comment (blasten) [inline packages/google_mobile_ads/ios/Tests/FLTNativeAdTest.m:68]: which `customOptions` is `Null`? `customOptions:nil` is used above.

File: packages/google_mobile_ads/ios/Tests/FLTRewardedAdTest.m [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+  OCMStub([mockReward type]).andReturn(@"type");
+  OCMStub([rewardedClassMock adReward]).andReturn(mockReward);
+  OCMStub([rewardedClassMock presentFromRootViewController:[OCMArg any]
+                                  userDidEarnRewardHandler:[OCMArg any]])
+  .andDo(^(NSInvocation *invocation) {
+    GADUserDidEarnRewardHandler rewardHandler;
+    [invocation getArgument:&rewardHandler atIndex:3];
+    rewardHandler();
+  });
+  // Prevent cals to setServerSideVerificationOptions if options is nil.
```
Comment (blasten) [inline packages/google_mobile_ads/ios/Tests/FLTRewardedAdTest.m:113]: nit: calls?

**Review by blasten — APPROVED** [review]
LGTM

File: packages/google_mobile_ads/ios/Tests/FLTRewardedAdTest.m (hunk 2 above)
Comment (jjliu15) [reply packages/google_mobile_ads/ios/Tests/FLTRewardedAdTest.m:113]: Done

File: packages/google_mobile_ads/ios/Tests/FLTNativeAdTest.m (hunk 1 above)
Comment (jjliu15) [reply packages/google_mobile_ads/ios/Tests/FLTNativeAdTest.m:69]: Updated the test method to take `customOptions` as an arg, and one fo the test methods to pass in `NULL`.

**Comment by jjliu15** [comment]
This has been merged into `major_version_all_changes`

**Pull request closed without merging**